paloma_sdk/client/lcd/api/cw721.py

```diff
@@ -0,0 +1,424 @@
+                "token_id": token_id,
+                "owner": owner,
+                "token_uri": token_uri,
+                "extension": None
+            }}
+
+        funds = Coins()
+        tx = await wallet.create_and_sign_tx(CreateTxOptions(
+            msgs=[MsgExecuteContract(
+                wallet.key.acc_address,
+                token,
+                execute_msg,
+                funds
+            )]
+        ))
+        result = await self._c.tx.broadcast(tx)
+        return result
+
+    async def approve(
+        self,
+        wallet: Wallet,
+        token: str,
+        spender: str,
+        token_id: str,
+        expires: Optional[dict] = None
+    ) -> BlockTxBroadcastResult:
+        """Allows operator to transfer / send the token from the owner's account.
+            If expiration is set, then this allowance has a time/height limit.
+        Args:
+            wallet (Wallet): CW721 owner wallet
+            token (str): CW721 token address
+            spender (str): the address who will get permission to the CW721 token
+            token_id (str): token id of the CW721 token
+            expires: (dict, optional) {"at_height": height(u64)} / {"at_time":timestamp(nanosecond, u64)}
+        Returns:
+            BlockTxBroadcastResult: Transaction Broadcast Result
+        """
+        execute_msg = {"approve": {
+            "spender": spender,
+            "token_id": token_id
+        }}
+
+        if expires is not None:
+            execute_msg["approve"]["expires"] = expires
+
+        funds = Coins()
+        tx = await wallet.create_and_sign_tx(CreateTxOptions(
+            msgs=[MsgExecuteContract(
+                wallet.key.acc_address,
+                token,
+                execute_msg,
+                funds
+            )]
+        ))
+        result = await self._c.tx.broadcast(tx)
+        return result
+
+    async def revoke(
+        self,
+        wallet: Wallet,
+        token: str,
+        spender: str,
+        token_id: str
+    ) -> BlockTxBroadcastResult:
+        """Remove previously granted Approval
+        Args:
+            wallet (Wallet): CW721 sender wallet
+            token (str): token address
+            spender (str): token approved address
+            token_id (str): CW721 token id
+        Returns:
+            BlockTxBroadcastResult: Transaction Broadcast Result
+        """
+        execute_msg = {"revoke": {
+            "spender": spender,
+            "token_id": token_id
+        }}
+
+        funds = Coins()
+        tx = await wallet.create_and_sign_tx(CreateTxOptions(
+            msgs=[MsgExecuteContract(
+                wallet.key.acc_address,
+                token,
+                execute_msg,
+                funds
+            )]
+        ))
+        result = await self._c.tx.broadcast(tx)
+        return result
+
+    async def approve_all(
+        self,
+        wallet: Wallet,
+        token: str,
+        operator: str,

```

Comment:
Is there a reason that we use `operator` here and not `spender` like we do for the approve() and revoke() functions?

Reply:
If we approve all tokens, the address can operate all the NFTs. That's why it is called an `operator`.
Please refer this.
https://github.com/CosmWasm/cw-nfts/blob/main/packages/cw721/README.md